File: python/client/libs/response_system/ResponseManager.py
class ResponseManager:
    def __init__(self):
        self.responsesWaitingList = dict()

    def waitResponse(self, response):
        self.responsesWaitingList.__setitem__(response.msgUUID, response)

    def onResponseReceived(self, msgUUID, message: str):
        if msgUUID in self.responsesWaitingList:
            self.responsesWaitingList.get(msgUUID).acceptReply(message)
            self.responsesWaitingList.pop(msgUUID)

File: golang/libs/response_system/response_manager.go
package response_system

import (
	"github.com/google/uuid"
)

type ResponseManager struct {
	responsesWaitingList map[uuid.UUID]*Response
}

func NewResponseManager() ResponseManager {
	return ResponseManager{
		responsesWaitingList: make(map[uuid.UUID]*Response),
	}
}

func (responseManager *ResponseManager) WaitResponse(response *Response) {
	responseManager.responsesWaitingList[response.UUID] = response
}

func (responseManager *ResponseManager) OnResponseReceived(UUID uuid.UUID, message string) {
	if r, exist := responseManager.responsesWaitingList[UUID]; exist {
		r.AcceptReply(message)
		delete(responseManager.responsesWaitingList, UUID)
	}
}
